src/MicrosoftAuthFlow.js

```diff
@@ -0,0 +1,150 @@
+            this.msa = new LiveTokenManager(this.options.authTitle, scopes, cachePaths.live)
+        } else { // Login with microsoftonline.com
+            const scopes = ['XboxLive.signin', 'offline_access']
+            this.msa = new MsaTokenManager(msalConfig, scopes, cachePaths.msa)
+        }
+
+        const keyPair = crypto.generateKeyPairSync('ec', { namedCurve: 'P-256' })
+        this.xbl = new XboxTokenManager(Authentication.XSTSRelyingParty, keyPair, cachePaths.xbl)
+        this.mca = new MinecraftTokenManager(cachePaths.bed)
+    }
+
+    static resetTokenCaches(cacheDir = minecraftFolderPath) {
+        const cache = path.join(cacheDir, 'nmp-cache')
+        try {
+            if (fs.existsSync(cache)) {
+                fs.rmdirSync(cache, { recursive: true })
+                return true
+            }
+        } catch (e) {
+            console.log('Failed to clear cache dir', e)
+            return false
+        }
+    }
+
+    async getMsaToken() {
+        if (await this.msa.verifyTokens()) {
+            debug('[msa] Using existing tokens')
+            return this.msa.getAccessToken().token
+        } else {
+            debug('[msa] No valid cached tokens, need to sign in')
+            const ret = await this.msa.authDeviceCode((response) => {
+                console.info('[msa] First time signing in. Please authenticate now:')
+                console.info(response.message)
+                if (this.codeCallback) this.codeCallback(response)
+            })
+
+            if (ret.account) {
+                console.info(`[msa] Signed in as ${ret.account.username}`)
+            } else { // We don't get extra account data here per scope
+                console.info('[msa] Signed in with Microsoft')
+            }
+
+            debug('[msa] got auth result', ret)
+            return ret.accessToken
+        }
+    }
+
+    async getXboxToken() {
+        if (await this.xbl.verifyTokens()) {
+            debug('[xbl] Using existing XSTS token')
+            return this.xbl.getCachedXstsToken().data
+        } else {
+            debug('[xbl] Need to obtain tokens')
+            return await retry(async () => {
+                const msaToken = await this.getMsaToken()
+                const ut = await this.xbl.getUserToken(msaToken, !this.options.authTitle)
+
+                if (this.options.authTitle) {
+                    const deviceToken = await this.xbl.getDeviceToken({ DeviceType: 'Nintendo', Version: '0.0.0' })
+                    const titleToken = await this.xbl.getTitleToken(msaToken, deviceToken)
+                    const xsts = await this.xbl.getXSTSToken(ut, deviceToken, titleToken)
+                    return xsts
+                } else {
+                    const xsts = await this.xbl.getXSTSToken(ut)
+                    return xsts
+                }
+            }, () => { this.msa.forceRefresh = true }, 2)
+        }
+    }
+
+    async getMinecraftToken(publicKey) {
```

Comment:
This can be renamed to getMinecraftBedrockToken() or something similar. Or the game target could be specified in the constructor and token obtained via getFinalToken/getGameToken().